### PR TITLE
XSD is stricter

### PR DESCRIPTION
**Pull request by yegor256**
see #3549

key changes:

- `<metas/>` must contain at least one `<meta/>`
- `<sheets/>` must contain at least one `<sheet/>`
- `<errors/>` must contain at least one `<error/>`
- `<comments/>` must contain at least one `<comment/>`
- `<listing/>` can't be empty
- `<license/>` can't be empty
- `<comment/>` can't be empty
- `<error/>` can't be empty
- `<sheet/>` can't be empty
- `/program/metas/meta/part` can't be empty
- `/program/metas/meta/head` must be `[a-z]+[a-z0-9]*`
- attributes `@dob`, `@revision`, `@version`, `@time` are optional at the `/program`
- XSD has more documentation blocks

### CODE REVIEW

**Comment by yegor256** [comment]
@maxonfjvipon please, check this one

File: eo-parser/src/main/resources/org/eolang/parser/cti-adds-errors.xsl [hunk 1]
```diff
@@ -22,30 +22,41 @@ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
...
+    <xsl:copy>
+      <xsl:apply-templates select="node()|@*"/>
+      <xsl:if test="//o[@base='cti']">
+        <errors>
+          <xsl:apply-templates select="//o[@base='cti']" mode="create"/>
+        </errors>
+      </xsl:if>
+    </xsl:copy>
+  </xsl:template>
+  <xsl:template match="//o[@base='cti']" mode="create">
```
Comment (maxonfjvipon) [inline eo-parser/src/main/resources/org/eolang/parser/cti-adds-errors.xsl:46]: @yegor256 I think just `o[@base='cti']` may work here

File: eo-maven-plugin/src/test/resources/org/eolang/maven/print/xmir/main.xmir [hunk 2]
```diff
@@ -58,4 +54,4 @@ SOFTWARE.
       </o>
     </o>
   </objects>
-</program>
\ No newline at end of file
+</program>
```
Comment (maxonfjvipon) [inline eo-maven-plugin/src/test/resources/org/eolang/maven/print/xmir/main.xmir:57]: @yegor256 I believe this change is unnecessary

File: eo-maven-plugin/src/test/resources/org/eolang/maven/print/xmir/inner/b.xmir [hunk 3]
```diff
@@ -65,4 +61,4 @@ SOFTWARE.
       </o>
     </o>
   </objects>
-</program>
\ No newline at end of file
+</program>
```
Comment (maxonfjvipon) [inline eo-maven-plugin/src/test/resources/org/eolang/maven/print/xmir/inner/b.xmir:64]: @yegor256 I believe this change is unnecessary

File: eo-maven-plugin/src/test/resources/org/eolang/maven/print/xmir/inner/a.xmir [hunk 4]
```diff
@@ -59,4 +55,4 @@ SOFTWARE.
       </o>
     </o>
   </objects>
-</program>
\ No newline at end of file
+</program>
```
Comment (maxonfjvipon) [inline eo-maven-plugin/src/test/resources/org/eolang/maven/print/xmir/inner/a.xmir:58]: @yegor256 I believe this change is unnecessary

File: eo-maven-plugin/src/test/resources/org/eolang/maven/print/xmir/bool-tests.xmir [hunk 5]
```diff
@@ -2619,4 +2616,4 @@ SOFTWARE.
       </o>
     </o>
   </objects>
-</program>
\ No newline at end of file
+</program>
```
Comment (maxonfjvipon) [inline eo-maven-plugin/src/test/resources/org/eolang/maven/print/xmir/bool-tests.xmir:2619]: @yegor256 I believe this change is unnecessary

File: eo-maven-plugin/src/test/resources/org/eolang/maven/main.xmir [hunk 6]
```diff
@@ -50,4 +48,4 @@ SOFTWARE.
          <o base=".print" line="4" method="" pos="19"/>
       </o>
    </objects>
-</program>
\ No newline at end of file
+</program>
```
Comment (maxonfjvipon) [inline eo-maven-plugin/src/test/resources/org/eolang/maven/main.xmir:51]: @yegor256 I believe this change is unnecessary

**Review by maxonfjvipon — CHANGES_REQUESTED** [review]

File: eo-maven-plugin/src/test/resources/org/eolang/maven/main.xmir (hunk 6 above)
Comment (yegor256) [reply eo-maven-plugin/src/test/resources/org/eolang/maven/main.xmir:51]: @maxonfjvipon in general, it's better to end every text file with EOL: https://stackoverflow.com/questions/729692/why-should-text-files-end-with-a-newline

Comment (maxonfjvipon) [reply eo-maven-plugin/src/test/resources/org/eolang/maven/main.xmir:51]: @yegor256 this means that you added EOL or deleted?
image

Comment (yegor256) [reply eo-maven-plugin/src/test/resources/org/eolang/maven/main.xmir:51]: @maxonfjvipon added

**Comment by 0crat** [comment]
@yegor256 Thanks for your contribution! Here's the breakdown: +4 base points, -8 for 757 hits-of-code (exceeding 200), -16 for no code review, and +24 to meet the minimum reward. While the large code size and lack of review impacted your score, we appreciate your effort. Remember, [quality over quantity](https://www.yegor256.com/2018/03/06/speed-vs-quality.html) is key. Your current balance stands at [-82](https://www.eolang.org/zerocracy/objectionary-vitals.html). Keep contributing, and focus on smaller, reviewed changes for better rewards!

**Pull request merged**